File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = prices[0];
        int maxProfit = 0;
        int n = prices.length;
        for(int i = 1; i < n; i++){
            if(buy < prices[i]){
                maxProfit = Math.max(maxProfit, prices[i] - buy);
            }
            buy = Math.min(buy, prices[i]);
        }

        return maxProfit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        buy = prices[0]
        maxProfit  = 0
        for i in range(len(prices)):
            if buy > prices[i]:
                buy = prices[i]
            
            maxProfit = max(maxProfit, prices[i] - buy)
        
        return maxProfit